src/puppetlabs/ring_middleware/core.clj

```diff
@@ -28,3 +29,21 @@
               (= request-method :put))
             (assoc-in response [:headers "cache-control"] "private, max-age=0, no-cache")
             response)))))
+
+(defn cert->principal
+  "Given an ssl cert, extract and return its X500Principal."
+  [cert]
+  (.getSubjectX500Principal cert))
```

Comment:
@camlow325 I'm working on a PR against jvm-ca (adding the new function was obviously trivial but it'll probably take me a day to figure out how to test it).

Once that PR is up against jvm-ca, is it okay to have this PR hang out until there is a new jvm-ca release with that PR merged?